src/pytom_tm/io.py

```diff
@@ -100,18 +100,33 @@ def __call__(self, parser, namespace, values: str, option_string: Optional[str]
         setattr(namespace, self.dest, read_dose_file(file_path))
 
 
-class ParseDefocusFile(argparse.Action):
-    """argparse.Action subclass to read a defocus file, either from IMOD which adheres to their file format,
-    or a txt file containing per line the defocus of each tilt."""
-    def __call__(self, parser, namespace, values: str, option_string: Optional[str] = None):
-        file_path = pathlib.Path(values)
-        if not file_path.exists():
-            parser.error("{0} provided defocus file does not exist".format(option_string))
-        allowed_suffixes = ['.defocus', '.txt']
-        if file_path.suffix not in allowed_suffixes:
-            parser.error("{0} provided defocus file does not have the right suffix, "
-                         "allowed are: {1}".format(option_string, ', '.join(allowed_suffixes)))
-        setattr(namespace, self.dest, read_defocus_file(file_path))
+class ParseDefocus(argparse.Action):
```

Comment:
this added function is untested, please add a test

Reply:
The input is tested via the entry point. I don't want to spend time right investigating how to write a test around these custom actions.

Reply:
Hitting it with the entry point is perfectly fine!